### ISSUE TITLE
issue with avg_muteffects

### ISSUE THREAD

**Issue by bblarsen-sci** [dms-vep/dms-vep-pipeline#161]
I'm not exactly sure why this is happening as this part of the pipeline has not changed. However, when I run the pipeline I am consistently getting this error:
```python
Traceback (most recent call last):
  File "/fh/fast/bloom_j/computational_notebooks/blarsen/2023/Nipah_Malaysia_RBP_DMS/.snakemake/conda/7cff5b2182da175431979118ea011288_/bin/papermill", line 11, in <module>
    sys.exit(papermill())
             ^^^^^^^^^^^
  File "/fh/fast/bloom_j/computational_notebooks/blarsen/2023/Nipah_Malaysia_RBP_DMS/.snakemake/conda/7cff5b2182da175431979118ea011288_/lib/python3.11/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/fh/fast/bloom_j/computational_notebooks/blarsen/2023/Nipah_Malaysia_RBP_DMS/.snakemake/conda/7cff5b2182da175431979118ea011288_/lib/python3.11/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/fh/fast/bloom_j/computational_notebooks/blarsen/2023/Nipah_Malaysia_RBP_DMS/.snakemake/conda/7cff5b2182da175431979118ea011288_/lib/python3.11/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/fh/fast/bloom_j/computational_notebooks/blarsen/2023/Nipah_Malaysia_RBP_DMS/.snakemake/conda/7cff5b2182da175431979118ea011288_/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/fh/fast/bloom_j/computational_notebooks/blarsen/2023/Nipah_Malaysia_RBP_DMS/.snakemake/conda/7cff5b2182da175431979118ea011288_/lib/python3.11/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/fh/fast/bloom_j/computational_notebooks/blarsen/2023/Nipah_Malaysia_RBP_DMS/.snakemake/conda/7cff5b2182da175431979118ea011288_/lib/python3.11/site-packages/papermill/cli.py", line 250, in papermill
    execute_notebook(
  File "/fh/fast/bloom_j/computational_notebooks/blarsen/2023/Nipah_Malaysia_RBP_DMS/.snakemake/conda/7cff5b2182da175431979118ea011288_/lib/python3.11/site-packages/papermill/execute.py", line 122, in execute_notebook
    raise_for_execution_errors(nb, output_path)
  File "/fh/fast/bloom_j/computational_notebooks/blarsen/2023/Nipah_Malaysia_RBP_DMS/.snakemake/conda/7cff5b2182da175431979118ea011288_/lib/python3.11/site-packages/papermill/execute.py", line 234, in raise_for_execution_errors
    raise error
papermill.exceptions.PapermillExecutionError: 
---------------------------------------------------------------------------
Exception encountered at "In [10]":
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[10], line 15
     12     plot_kwargs["addtl_slider_stats"]["n_libraries"] = 1
     14 if "region" in sitenumbering_map.columns:
---> 15     df_to_plot = df_to_plot.merge(
     16         sitenumbering_map.rename(columns={"reference_site": "site"})[["site", "region"]]
     17     )
     18     plot_kwargs["site_zoom_bar_color_col"] = "region"
     20 if "addtl_tooltip_stats" not in plot_kwargs:

File /fh/fast/bloom_j/computational_notebooks/blarsen/2023/Nipah_Malaysia_RBP_DMS/.snakemake/conda/7cff5b2182da175431979118ea011288_/lib/python3.11/site-packages/pandas/core/frame.py:9848, in DataFrame.merge(self, right, how, on, left_on, right_on, left_index, right_index, sort, suffixes, copy, indicator, validate)
   9829 @Substitution("")
   9830 @Appender(_merge_doc, indents=2)
   9831 def merge(
   (...)
   9844     validate: str | None = None,
   9845 ) -> DataFrame:
   9846     from pandas.core.reshape.merge import merge
-> 9848     return merge(
   9849         self,
   9850         right,
   9851         how=how,
   9852         on=on,
   9853         left_on=left_on,
   9854         right_on=right_on,
   9855         left_index=left_index,
   9856         right_index=right_index,
   9857         sort=sort,
   9858         suffixes=suffixes,
   9859         copy=copy,
   9860         indicator=indicator,
   9861         validate=validate,
   9862     )

File /fh/fast/bloom_j/computational_notebooks/blarsen/2023/Nipah_Malaysia_RBP_DMS/.snakemake/conda/7cff5b2182da175431979118ea011288_/lib/python3.11/site-packages/pandas/core/reshape/merge.py:144, in merge(left, right, how, on, left_on, right_on, left_index, right_index, sort, suffixes, copy, indicator, validate)
    127 @Substitution("\nleft : DataFrame or named Series")
    128 @Appender(_merge_doc, indents=0)
    129 def merge(
   (...)
    142     validate: str | None = None,
    143 ) -> DataFrame:
--> 144     op = _MergeOperation(
    145         left,
    146         right,
    147         how=how,
    148         on=on,
    149         left_on=left_on,
    150         right_on=right_on,
    151         left_index=left_index,
    152         right_index=right_index,
    153         sort=sort,
    154         suffixes=suffixes,
    155         indicator=indicator,
    156         validate=validate,
    157     )
    158     return op.get_result(copy=copy)

File /fh/fast/bloom_j/computational_notebooks/blarsen/2023/Nipah_Malaysia_RBP_DMS/.snakemake/conda/7cff5b2182da175431979118ea011288_/lib/python3.11/site-packages/pandas/core/reshape/merge.py:737, in _MergeOperation.__init__(self, left, right, how, on, left_on, right_on, axis, left_index, right_index, sort, suffixes, indicator, validate)
    729 (
    730     self.left_join_keys,
    731     self.right_join_keys,
    732     self.join_names,
    733 ) = self._get_merge_keys()
    735 # validate the merge keys dtypes. We may need to coerce
    736 # to avoid incompatible dtypes
--> 737 self._maybe_coerce_merge_keys()
    739 # If argument passed to validate,
    740 # check if columns specified as unique
    741 # are in fact unique.
    742 if validate is not None:

File /fh/fast/bloom_j/computational_notebooks/blarsen/2023/Nipah_Malaysia_RBP_DMS/.snakemake/conda/7cff5b2182da175431979118ea011288_/lib/python3.11/site-packages/pandas/core/reshape/merge.py:1389, in _MergeOperation._maybe_coerce_merge_keys(self)
   1383     # unless we are merging non-string-like with string-like
   1384     elif (
   1385         inferred_left in string_types and inferred_right not in string_types
   1386     ) or (
   1387         inferred_right in string_types and inferred_left not in string_types
   1388     ):
-> 1389         raise ValueError(msg)
   1391 # datetimelikes must match exactly
   1392 elif needs_i8_conversion(lk.dtype) and not needs_i8_conversion(rk.dtype):

ValueError: You are trying to merge on int64 and object columns. If you wish to proceed you should use pd.concat
```

For some reason `df_to_plot['site']` is an int64 while `sitenumbering_map['site']` is an object. My site_numbering_map.csv file does contain a position at the very end of the gene (602a) where there is a very minor mutation to the stop codon that wasn't designed. I don't know if this is causing the issue, but I have been using the same file this entire time so I'm not sure why suddenly it would throw up this error. I looked at all the recent changes to this Jupyter notebook and script and don't see anything that stands out for why this error would suddenly be occurring. My GitHub repo is not current as I'm trying to get this new dms-vep-pipeline module up and running, but /computational_notebooks/blarsen/2023/Nipah_Malaysia_RBP/DMS contains the current files for this issue I am trying to resolve

**Comment by bblarsen-sci** [comment]
Maybe its an issue with the updated pandas version?

**Comment by jbloom** [comment]
Yes, it probably relates to updated `pandas` version, but I will look into how to fix it without reverting `pandas` version.

**Comment by jbloom** [comment]
@bblarsen-sci, I think this should be fixed in version 2.6.1.

If it isn't, re-open this issue and let me know the error.
